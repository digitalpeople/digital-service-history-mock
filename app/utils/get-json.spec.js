import fs from 'fs';
import getJson from './get-json';

jest.mock('fs', () => ({
  readFile: jest.fn(),
}));
jest.mock('./response-handler', () => () => {});

describe('getJson', () => {
  it('should append response Content-Type', () => {
    expect.assertions(1);

    const path = './test.json';
    const response = {
      append: jest.fn(),
    };

    getJson(path, response);

    expect(response.append.mock.calls[0][0]).toMatchSnapshot();
  });

  it('should call fs.readFile', () => {
    expect.assertions(1);

    const path = './test.json';
    const response = {
      append: jest.fn(),
    };

    getJson(path, response);

    expect(fs.readFile.mock.calls[0][0]).toMatchSnapshot();
  });
});
