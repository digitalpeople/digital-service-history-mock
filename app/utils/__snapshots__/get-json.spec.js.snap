// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getJson should append response Content-Type 1`] = `"Content-Type"`;

exports[`getJson should call fs.readFile 1`] = `"./test.json"`;
